Update to default k8s deployment docs (#22046)
gravitational/teleport: docs/pages/deploy-a-cluster/helm-deployments/kubernetes-cluster.mdx

Teleport can provide secure, unified access to your Kubernetes clusters. This guide will show you how to:

<ScopedBlock scope={["enterprise"]}>

> - Install the Teleport Enterprise License.

Reviewer: Is this a distinct task from deploying Teleport Enterprise in a Kubernetes cluster?

Author: Basically installing the license as a secret.  We can remove it if you find it confusing.

> - Install the Teleport Enterprise License.

Reviewer: ```suggestion
```
For me, this is more of a step toward deploying Teleport Enterprise on Kubernetes, rather than a standalone step

  ```
</Admonition>

## Troubleshooting

> If you are experiencing errors connecting with the Teleport Cluster check the proxy
> and auth pods status. A successful state should be both pods running as below:

Reviewer: ```suggestion
If you are experiencing errors connecting to the Teleport cluster, check the status of the Auth Service and Proxy Service pods. A successful state should show both pods running as below:
```

```code
$ kubectl get pods -n teleport-cluster
NAME                                READY   STATUS    RESTARTS   AGE
teleport-cluster-auth-5f8587bfd4-p5zv6    1/1     Running   0          48s
teleport-cluster-proxy-767747dd94-vkxz6   1/1     Running   0          48s
```

> If a Pod is Pending or not running completely use the `kubectl logs` and `kubectl describe` 

Reviewer 2: What does "not running completely" mean? Does Kubernetes have the concept of a pod that is only partially running?

Author: Updated language to describe the status or not ready which is the more typical language

Author: `0/1` and the Pod is running mean it isn't ready.

> If a Pod Status is Pending or not ready use the `kubectl logs` and `kubectl describe` 
> for that pod to check the status. The auth pod relies on being able to allocate a Persistent Volume Claim so that may
> put it in a Pending state if no Persistent Volume is available.

Reviewer: ```suggestion
If a pod's status is `Pending`, use the `kubectl logs` and `kubectl describe` commands 
for that pod to check the status. The Auth Service pod relies on being able to allocate a Persistent Volume Claim, and may enter a `Pending` state if no Persistent Volume is available.
```
